src/subdomains/generic/user/models/user/user.controller.ts

```diff
@@ -228,6 +231,34 @@ export class UserV2Controller {
     return user;
   }
 
+  @Put('addresses/:address')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.ACCOUNT))
+  @ApiOkResponse({ type: UserV2Dto })
+  async updateAddress(
+    @GetJwt() jwt: JwtPayload,
+    @Body() newAddress: UpdateAddressDto,
+    @Param('address') address: string,
+  ): Promise<UserV2Dto> {
+    return this.userService.updateAddress(jwt.account, address, newAddress);
+  }
+
+  @Delete('addresses/:address')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.ACCOUNT))
+  @ApiOkResponse()
+  async deleteAddress(@GetJwt() jwt: JwtPayload, @Param('address') address: string): Promise<void> {
+    return this.userService.deactivateUser(jwt.account, address);
+  }
+
+  @Delete()
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.ACCOUNT))
+  @ApiOkResponse()
+  async deleteAccount(@GetJwt() jwt: JwtPayload): Promise<void> {
+    return this.userService.deactivateUser(jwt.account);
+  }
```

Comment:
Could we change the Guard in the deprecated endpoint above to `Account`?
In the moment this endpoint would not be reached because there is the deprecated Delete endpoint above. Else we should add a `TODO: remove` on the above Delete endpoint or?

Reply:
This is in the v2 controller, so there should be no clashes.